StockTrader/src/main/java/com/restResource/StockTrader/controller/BuyTriggerController.java

```diff
@@ -0,0 +1,169 @@
+import com.restResource.StockTrader.entity.logging.ErrorEventLog;
+import com.restResource.StockTrader.entity.logging.UserCommandLog;
+import com.restResource.StockTrader.repository.AccountRepository;
+import com.restResource.StockTrader.repository.BuyTriggerRepository;
+import com.restResource.StockTrader.service.BuyTriggerService;
+import com.restResource.StockTrader.service.LoggingService;
+import com.restResource.StockTrader.service.QuoteService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/buyTrigger")
+public class BuyTriggerController {
+
+    private AccountRepository accountRepository;
+
+    private LoggingService loggingService;
+
+    private BuyTriggerRepository buyTriggerRepository;
+
+    private BuyTriggerService buyTriggerService;
+
+    public BuyTriggerController(
+            BuyTriggerRepository buyTriggerRepository,
+            LoggingService loggingService,
+            BuyTriggerService buyTriggerService,
+            AccountRepository accountRepository) {
+
+        this.buyTriggerRepository = buyTriggerRepository;
+        this.loggingService = loggingService;
+        this.buyTriggerService = buyTriggerService;
+        this.accountRepository = accountRepository;
+    }
+
+    @PostMapping(path = "/amount")
+    public @ResponseBody
+    HttpStatus createTriggerAmount(
+            @RequestParam String userId,
+            @RequestParam String stockSymbol,
+            @RequestParam int stockAmount,
+            @RequestParam int transactionNum) {
+
+
+        loggingService.logUserCommand(
+                UserCommandLog.builder()
+                        .command(CommandType.SET_BUY_AMOUNT)
+                        .username(userId)
+                        .stockSymbol(stockSymbol)
+                        .transactionNum(transactionNum)
+                        .build());
+
+        if (stockAmount <= 0) {
+            loggingService.logErrorEvent(
+                    ErrorEventLog.builder()
+                            .command(CommandType.SET_BUY_AMOUNT)
+                            .username(userId)
+                            .stockSymbol(stockSymbol)
+                            .transactionNum(transactionNum)
+                            .errorMessage("The amount parameter must be greater than zero")
+                            .build());
+            //FIXME do we want to throw the error or send a HttpStatus message?
+            throw new IllegalArgumentException(
+                    "The amount parameter must be greater than zero.");
+        }
+
+        Optional<BuyTrigger> stockBuyTriggerStatus = buyTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+
+        if (stockBuyTriggerStatus.isPresent()) {
+            Integer cost = stockBuyTriggerStatus.get().getStock_cost();
```

Comment:
Isn't stockCost the trigger price? I think you wanted stockAmount here (which also shouldn't be null so you don't have to check for that